### ISSUE TITLE
Is a noop. Returns (clang-format: success) in mini buffer

### ISSUE THREAD

**Issue by appetrosyan** [sonatard/clang-format#8]
Steps to reproduce open any file with extension `.cpp`. and run `clang-format-buffer`. 

* expected behaviour: 
file changes

* actual behaviour
noop.

**Comment by Paethon** [comment]
Yes, same for me. Worked for years, but currently it seems to be broken (just doesn't do anything)

**Comment by hojoon-lee** [comment]
I had the same issue, and managed to fix the problem by replacing clang-format.el from melpa with the one in the newest clang installation.

**Comment by philipp-classen** [comment]
Finally, I figured it out. It only works if you have a .clang-format file:

https://emacs.stackexchange.com/a/55650/8261

Otherwise, it will use the fallback style. I changed it now to:

    Clang Format Fallback Style: llvm

By default, it is `none`, which explains why nothing changed.

**Comment by haifenghan** [comment]
Thanks @philipp-classen , i-t works for me.

**Comment by haifenghan** [comment]
Copy from the stackexchange:

(*) To generate a .clang-format file, you can use:

`clang-format -style=llvm -dump-config > .clang-format`
